source/camera.ts

```diff
@@ -365,4 +392,37 @@ export class Camera {
         this._altered = status;
     }
 
+    /**
+     * Returns a deep copy of the camera.
+     */
+    public copy(): Camera {
+        const copiedCamera = new Camera(this.eye, this.center, this.up);
+        copiedCamera.altered = this.altered;
+        copiedCamera.fovy = this.fovy;
+        copiedCamera.near = this.near;
+        copiedCamera.far = this.far;
+        copiedCamera.viewport = [this.viewport[0], this.viewport[1]];
+        copiedCamera.aspect = this.aspect;
+        copiedCamera.postViewProjection = this.postViewProjection;
```

Comment:
This matrix is passed by reference. As the camera is referred to as a *copied* camera, I assume that the internal projection matrix has to be copied, too.

Reply:
While reviewing the next method (which actually copies the projection matrix), I imagined the following implementation for `copy`:
```typescript
const copiedCamera = new Camera(/* ... */);
this.copyAllValues(copiedCamera);
return copiedCamera;
```

Reply:
Thanks there was definitely some code duplication.